### DISCUSSION TITLE
Replica Setup and Routing Best Practices

### DISCUSSION THREAD

**Discussion by Git-Me-Some-Hub** [Security-Onion-Solutions/securityonion#4998] [Unsupported Versions]
ISO - 2.3.61-Stenodocker

Long term I am looking to have hot/warm, replicas, and snapshot curator actions and node. I see more than enough ways to do this out of the internet via a pure ES setup, but I am not 100% sure how this works in the SO environment. The reason is I am not sure what I can just put in the elasticsearch.yml file and what I really should have in the Global/Minion files. 

Do I need to change the cluster name from the individual hostnames to a shared cluster? Do I need to assign one or more of the nodes as masters, do I need to list a discovery node? 

The reason I am asking this is that I have some shards that are unassigned because they are duplicates of what is already present. From research it is because I set replica: 1 and made a curator action, but it is trying to save that replica shard to the same node that originally has it. I tried a command out there to reroute/allocate those shards, but it did not work. 

I guess I could try and delete them?

`curl -k -XPUT -H'Content-Type: application/json' https://localhost:9200/_cluster/reroute?pretty' -d '{
"commands" : [ {
"allocate_empty_primary" :
{
"index" : "so-ossec-2021.07.21", "shard" : 1,
"node" : "<clustername>",
"accept_data_loss" : true
}
}
]
}'`

Thanks for reading.

**Comment by TOoSmOotH** [comment]
To switch to true clustering is not trivial. In order to do this, you will first need to rename all of your indices to something unique. Then you will need to re-run setup on all nodes to enable true clustering. Once that is done you will need to pull in the dangling indices into the new true cluster. Once you have a true cluster then replicas will be honored and will spread across the cluster.

**Reply by Git-Me-Some-Hub** [reply]
Got it, thanks. This is much more than I was initially expecting, but good to have some set of tasks I need to google to start down that path.

**Reply by 3isenHeiM** [reply]
@Git-Me-Some-Hub did you finally managed to perform this switch ? I'm also interested...
